components/org.wso2.carbon.database.utils/src/main/java/org/wso2/carbon/database/utils/jdbc/NamedJdbcTemplate.java

```diff
@@ -258,36 +259,58 @@ public int executeUpdate(String query) throws DataAccessException {
+     * @param namedQueryFilter  Query filter to prepared statement parameter binding.
+     * @param bean              the Domain object to be inserted/updated.
+     * @param fetchInsertedId   Fetch inserted ID.
+     * @param <T>               return type of the object.
      */
     public <T> int executeInsert(String query, NamedQueryFilter namedQueryFilter, T bean, boolean fetchInsertedId)
             throws DataAccessException {
 
-        try (Connection connection = dataSource.getConnection();
-             NamedPreparedStatement namedPreparedStatement = new NamedPreparedStatement(connection, query)) {
+        return executeInsert(query, namedQueryFilter, bean, fetchInsertedId, ID);
+    }
+
+    /**
+     * Executes the jdbc insert/update query.
+     *
+     * @param query                     The SQL for insert/update.
+     * @param namedQueryFilter          Query filter to named prepared statement parameter binding.
+     * @param bean                      the Domain object to be inserted/updated.
+     * @param fetchInsertedId           Fetch inserted ID.
+     * @param autoGenerateIdColumnName  Name of auto generate ID column.
+     * @param <T>                       return type of the object.
+     */
+    public <T> int executeInsert(String query, NamedQueryFilter namedQueryFilter, T bean, boolean fetchInsertedId,
+                                 String autoGenerateIdColumnName)
+            throws DataAccessException {
+
+        try (Connection connection = dataSource.getConnection()) {
             int resultId;
             if (fetchInsertedId) {
-                doInternalUpdate(namedQueryFilter, namedPreparedStatement);
-                logDebug("Mapping generated key (Auto Increment ID) to the object");
-                try (ResultSet generatedKeys = namedPreparedStatement.getGeneratedKeys()) {
-                    if (generatedKeys.next()) {
-                        resultId = generatedKeys.getInt(1);
-                        logDebug("Newly inserted ID (Auto Increment ID) is {} for the bean {} ", resultId,
-                                bean);
-                    } else {
-                        throw new SQLException(JdbcConstants.ErrorCodes.ERROR_CODE_AUTO_GENERATED_ID_FAILURE.
-                                getErrorMessage());
+                try (NamedPreparedStatement namedPreparedStatement = new NamedPreparedStatement(connection, query,
+                        autoGenerateIdColumnName)) {
+                    doInternalUpdate(namedQueryFilter, namedPreparedStatement);
+                    logDebug("Mapping generated key (Auto Increment ID) to the object");
+                    try (ResultSet generatedKeys = namedPreparedStatement.getGeneratedKeys()) {
+                        if (generatedKeys.next()) {
+                            resultId = generatedKeys.getInt(1);
+                            logDebug("Newly inserted ID (Auto Increment ID) is {} for the bean {} ", resultId,
```

Comment:
Full stop at the end of the log message

Reply:
updated 81e310b1cff1b6d5a106bd6ae81fd5137d85bcdf